pyaim/aimccp.py

```diff
@@ -0,0 +1,63 @@
+import json
+import ssl
+import urllib.parse
+import urllib.request
+
+
+class CCPPasswordREST(object):
+
+    # Runs on Initialization
+    def __init__(self, base_uri, verify=True):
+
+        # Declare Init Variables
+        self._base_uri = base_uri.rstrip('/')   # Example: https://pvwa.cyberarkexample.com
+        
+        if verify is False:
+            self._context = ssl.create_default_context()
+            self._context.check_hostname = False
+            self._context.verify_mode = ssl.CERT_NONE
+        else:
+            self._context = ssl.create_default_context()
+
+
+    # Checks that the AIM Web Service is available
+    def check_service(self):
+        try:
+            conn = urllib.request.urlopen(self._base_uri + '/AIMWebService/v1.1/aim.asmx', context=self._context)
+            status_code = conn.getcode()
+            conn.close()
```

Comment:
Very likely, the connection uses a context manager so you can use this instead of what you have here:
```
with <that long urllib line> as conn:
   <do stuff with result>
```
It should auto-close the context when you get out of the indented area